src/transactions/TransactionFrame.cpp

```diff
@@ -751,6 +751,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                 newMeta.v2().txChangesAfter = ltxAfter.getChanges();
                 ltxAfter.commit();
             }
+            else if (ledgerVersion >= 14)
+            {
+                auto delta = ltxTx.getDelta();
+                for (auto const& kv : delta.entry)
+                {
+                    auto glk = kv.first;
+                    if (glk.type() == GeneralizedLedgerEntryType::SPONSORSHIP)
```

Comment:
I think we need `SPONSORSHIP_COUNTER` to implement `ACCOUNT_MERGE` and `SPONSOR_FUTURE_RESERVES` efficiently. In those places, we need reverse-lookup to determine if an account is a sponsor rather than is sponsored. This allows us to prevent bad things like:
- merging an account that is-sponsoring-future-reserves-for another account
- recursive sponsorship where account is-sponsoring-future-reserves-for an account that is-sponsoring-future-reserves-for some other account.